pkg/operator/ceph/object/realm/controller.go

```diff
@@ -0,0 +1,240 @@
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Reconcile reads that state of the cluster for a CephObjectStoreRealm object and makes changes based on the state read
+// and what is in the CephObjectStoreRealm.Spec
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileObjectStoreRealm) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// workaround because the rook logging mechanism is not compatible with the controller-runtime loggin interface
+	reconcileResponse, err := r.reconcile(request)
+	if err != nil {
+		logger.Errorf("failed to reconcile %v", err)
+	}
+
+	return reconcileResponse, err
+}
+
+func (r *ReconcileObjectStoreRealm) reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the CephObjectStoreRealm instance
+	cephObjectStoreRealm := &cephv1.CephObjectStoreRealm{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, cephObjectStoreRealm)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debug("CephObjectStoreRealm resource not found. Ignoring since object must be deleted.")
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrap(err, "failed to get CephObjectStoreRealm")
+	}
+
+	// The CR was just created, initializing status fields
+	if cephObjectStoreRealm.Status == nil {
+		cephObjectStoreRealm.Status = &cephv1.Status{}
+		cephObjectStoreRealm.Status.Phase = k8sutil.Created
+		err := opcontroller.UpdateStatus(r.client, cephObjectStoreRealm)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to set status")
+		}
+	}
+
+	// Make sure a CephCluster is present otherwise do nothing
+	_, isReadyToReconcile, cephClusterExists, reconcileResponse := opcontroller.IsReadyToReconcile(r.client, r.context, request.NamespacedName, controllerName)
+	if !isReadyToReconcile {
+		// This handles the case where the Ceph Cluster is gone and we want to delete that CR
+		//
+		// Also, only remove the finalizer if the CephCluster is gone
+		// If not, we should wait for it to be ready
+		// This handles the case where the operator is not ready to accept Ceph command but the cluster exists
+		if !cephObjectStoreRealm.GetDeletionTimestamp().IsZero() && !cephClusterExists {
+			// Remove finalizer
+			err = opcontroller.RemoveFinalizer(r.client, cephObjectStoreRealm)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrap(err, "failed to remove finalizer")
+			}
+
+			// Return and do not requeue. Successful deletion.
+			return reconcile.Result{}, nil
+		}
+		return reconcileResponse, nil
+	}
+
+	// Set a finalizer so we can do cleanup before the object goes away
+	err = opcontroller.AddFinalizerIfNotPresent(r.client, cephObjectStoreRealm)
```

Comment:
When the realm CR is deleted, are we ever going to remove the realm from ceph? If not, we don't need a finalizer.